EIPS/eip-7718.md

```diff
@@ -0,0 +1,418 @@
+
+Currently defined `angelfood` protocol identifiers:
+
+- `0x504A`: Execution State Network
+- `0x504B`: Execution History Network
+- `0x504C`: Beacon Chain Network
+- `0x504D`: Execution Canonical Transaction Index Network
+- `0x504E`: Execution Verkle State Network
+- `0x504F`: Execution Transaction Gossip Network
+
+### Content Keys and Content IDs
+
+Content keys are used to request or offer specific content data. As such the content key and content data can be represented as a key:value pair.
+
+Content keys are passed as byte strings to the messages defined in the Portal wire protocol. How they are encoded is defined per content network specification.
+
+Content IDs are derived from the content keys and are used to identify where the content is located in the network. The derivation is defined per content network specification.
+
+#### SHA256 Content ID Derivation Function
+
+The SHA256 Content ID derivation function is defined as:
+
+```
+content_id = sha256(content_key)
+```
+
+### Nodes and Node IDs
+
+Nodes in the portal network are represented by their [EIP-778 Ethereum Node Record (ENR)](./eip-778.md) from the Discovery v5 network. A node's `node-id` is derived according to the node's identity scheme, which is specified in the node's ENR. A node's `node-id` represents its address in the DHT.  Node IDs are interchangeable between 32 byte identifiers and 256 bit integers.
+
+### Request - Response Messages
+
+The messages in the protocol are transmitted using the `TALKREQ` and `TALKRESP` messages from the base [Node Discovery Protocol](https://github.com/ethereum/devp2p/blob/56a498ee34ee0fb69ffd33dda026d632af4c4048/discv5/discv5-wire.md#talkreq-request-0x05).
+
+All messages in the protocol have a request-response interaction:
+
+- Request messages **MUST** be sent using a `TALKREQ` message.
+- Response messages **MUST** be sent using the corresponding `TALKRESP` message.
+
+All messages are encoded as an [SSZ Union](https://github.com/ethereum/consensus-specs/blob/04f5ec595d78c0e3e43794fb7644c18f2584770d/ssz/simple-serialize.md#union) type.
+
+```
+message = Union[ping, pong, find_nodes, nodes, find_content, content, offer, accept]
+serialized_message = serialize(message)
+```
+
+The `serialized_message` is the payload passed to the `request` field of the `TALKREQ` message or the `reponse` field of the `TALKRESP` message.
+
+The type values for the `Union` are the SSZ Containers specified per message type.
+
+The transmission of `content` data that is too large to fit a single packet is done over [uTP](../assets/eip-7718/bep_0029-rst_post.pdf).
+
+#### Aliases
+
+For convenience we alias:
+
+- `ByteList` to `List[uint8, max_length=2048]`
+
+#### Message Types
+
+##### Ping (0x00)
+
+Request message to check if a node is reachable, communicate basic information about our node, and request basic information about the recipient node.
+
+```
+selector     = 0x00
+ping         = Container(enr_seq: uint64, custom_payload: ByteList)
```

Comment:
We probably want to call this `ByteList[2048]` as the SSZ spec already defines `ByteList[N]`. 

(I think we also might want to lower it in case we want to be strict here in accordance to the discv5 packet limit + protocol overhead.)